packages/eslint-plugin-react-hooks/__tests__/ESLintRulesOfHooks-test.js

```diff
@@ -406,6 +406,93 @@ const tests = {
         const [myState, setMyState] = useState(null);
       }
     `,
+    `
+      // Valid because functions created with useEvent can be called in a useEffect.
+      function MyComponent({ theme }) {
+        const onClick = useEvent(() => {
+          showNotification(theme);
+        });
+        useEffect(() => {
+          onClick();
+        });
+      }
+    `,
+    `
+      // Valid because functions created with useEvent can be called in closures.
+      function MyComponent({ theme }) {
+        const onClick = useEvent(() => {
+          showNotification(theme);
+        });
+        return <Child onClick={() => onClick()}></Child>;
+      }
+    `,
+    `
+      // Valid because functions created with useEvent can be called in closures.
+      function MyComponent({ theme }) {
+        const onClick = useEvent(() => {
+          showNotification(theme);
+        });
+        const onClick2 = () => { onClick() };
+        return <Child onClick={onClick2}></Child>;
+      }
+    `,
+    `
+      // Valid because functions created with useEvent can be passed by reference in useEffect.
+      function MyComponent({ theme }) {
+        const onClick = useEvent(() => {
+          showNotification(theme);
+        });
+        useEffect(() => {
+          let id = setInterval(onClick, 100);
+          return () => clearInterval(onClick);
+        }, []);
+      }
+    `,
+    `
+      // Valid because onClick is invoked via call or apply
+      function MyComponent({ theme }) {
+        const onClick = useEvent(() => {
+          showNotification(theme);
+        });
+        return <>
+          <Child onClick={onClick.call()} />
+          <Child onClick={onClick.apply()} />
+        </>
+      }
+    `,
+    `
+      // Valid because onClick is invoked, and is created in MyComponent's environment prior to
+      // invocation
+      function MyComponent({ theme }) {
+        function Child() {
```

Comment:
It's not valid React but it is valid JavaScript, so I was erring on the side of catching more cases rather than only the happy path. I can remove it, which does simplify a lot of the code

Reply:
What happens if your remove the special handling? What does the rule do?